src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -987,6 +987,31 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
                 node.terminal_pressure(pressure_item.getSIDouble(0));
 
             if (as_choke) {
+                //Wells belong to a group with autochoke enabled are to be run on a common THP and guide rate should be ignored
+
+                const auto& group = this->getGroup(name, handlerContext.currentStep);
+                for (const std::string& wellName : group.wells()) {
+                    auto well = this->snapshots.back().wells.get(wellName);
+
+                    //Let the wells be operating on a THP Constraint 
+                    auto properties = std::make_shared<Well::WellProductionProperties>(well.getProductionProperties());
+                    properties->clearControls();
```

Comment:
It is just some comments for discussion. 

I **assume** that we should not clear other controls. We only need to make the well is not available from group control.  It is something I also learned recently through discussion with you, `GRUP` control looks like is there but we use whether it is available to group control to control whether it is under `GRUP` control. 

**Maybe** if the well is set to be `GRUP` control, we can set it to be under `THP` control.

Reply:
Indeed, we should not clear all. It is now changed to dropping only `GRUP` control and setting `THP` control for wells part of an auto-choke group.